internal/host/host.go

```diff
@@ -954,23 +954,29 @@ func (m *Manager) ResetPendingUserAction(ctx context.Context, h *models.Host, db
 	return nil
 }
 
-func (m *Manager) GetStagesByRole(role models.HostRole, isbootstrap bool, isSNO bool) []models.HostStage {
-	if isbootstrap || role == models.HostRoleBootstrap {
+func (m *Manager) GetStagesByRole(h *models.Host, isSNO bool) []models.HostStage {
```

Comment:
Now that this function is correct, I think we can remove https://github.com/openshift/assisted-service/blob/589d1530344dff8b803e1c8a79ecab4bf0170fb9/internal/host/host.go#L576-L579 as it is redundant.